Dockerfile

```
@@ -79,37 +76,49 @@ RUN conan export conan/mqtt_cpp \
 FROM os AS release
 
 LABEL author="mtconnect" description="Docker image for the latest Production MTConnect C++ Agent"
+ARG BIN_DIR=/usr/bin
+ARG MTCONNECT_DIR=/etc/mtconnect
+
+ENV MTCONNECT_DIR=$MTCONNECT_DIR
 
 # install ruby for simulator
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y ruby zip
 
 # change to a new non-root user for better security.
 # this also adds the user to a group with the same name.
 # --create-home creates a home folder, ie /home/<username>
 RUN useradd --create-home agent
 USER agent
+WORKDIR /home/agent
 
 # install agent executable
-COPY --chown=agent:agent --from=build /root/agent/build/bin/agent /usr/local/bin/
+COPY --chown=agent:agent --from=build /root/agent/*.zip .
 
 # copy data to /etc/mtconnect
-COPY --chown=agent:agent --from=build /root/agent/schemas /etc/mtconnect/schemas
-COPY --chown=agent:agent --from=build /root/agent/simulator /etc/mtconnect/simulator
-COPY --chown=agent:agent --from=build /root/agent/styles /etc/mtconnect/styles
+RUN unzip *.zip
+
+ENV ZIP_DIR=/home/agent/agent-*-Linux
+
+USER root
+RUN cp $ZIP_DIR/bin/* $BIN_DIR
+RUN mkdir -p $MTCONNECT_DIR && chown -R agent:agent $MTCONNECT_DIR
+
+USER agent
+RUN cp -r $ZIP_DIR/schemas $ZIP_DIR/simulator $ZIP_DIR/styles $ZIP_DIR/demo $MTCONNECT_DIR/
 
 # expose port
 EXPOSE 5000
 
-WORKDIR /home/agent
+ENV DEMO_DIR=$MTCONNECT_DIR/demo/agent
 
 # default command - can override with docker run or docker-compose command.
 # this runs the adapter simulator and the agent using the sample config file.
-# note: must use shell form here instead of exec form, since we're running 
+# note: must use shell form here instead of exec form, since we're running
 # multiple statements using shell commands (& and &&).
 # see https://stackoverflow.com/questions/46797348/docker-cmd-exec-form-for-multiple-command-execution
-CMD /usr/bin/ruby /etc/mtconnect/simulator/run_scenario.rb -l \
-  /etc/mtconnect/simulator/VMC-3Axis-Log.txt & \
-  cd /etc/mtconnect/simulator && mtcagent run agent.cfg
+CMD /usr/bin/ruby $MTCONNECT_DIR/simulator/run_scenario.rb -p 7879 -l $DEMO_DIR/mazak.txt & \
+    /usr/bin/ruby $MTCONNECT_DIR/simulator/run_scenario.rb -p 7878 -l $DEMO_DIR/okuma.txt & \
+    cd $DEMO_DIR && mtcagent run agent.cfg
```

1. During my tests, `conan` built `agent` binary, not `mtcagent`. You might have changed some code to build `mtcagent`, however, I want to verify this before moving forward.
2. Does the agent required to change to `DEMO_DIR` before running? or is this only because of `agent.cfg` is in that folder? Won’t it work if we run it as follows?

```suggestion
    mtcagent run "$DEMO_DIR/agent.cfg"
```

---

> [@wsobel: [src](https://github.com/mtconnect/cppagent/pull/303#issuecomment-1666964450)]
>
> 1. The option -o agent_prefix=mtc prefixes the library and the agent binary.
>
>2. Will test tomorrow. Late now in Fr.

1. Oh, thanks! :pray:
2. Okay. BTW, I like in Slovakia, therefore the time is the same. :wink: Night! :wink: